crates/exex/exex/src/backfill.rs

```diff
@@ -195,38 +197,124 @@ where
     }
 }
 
+impl<E, DB, P> BackfillJob<E, DB, P> {
+    /// Converts the backfill job into a single block backfill job.
+    pub fn into_single_blocks(self) -> SingleBlockBackfillJob<E, DB, P> {
+        self.into()
+    }
+}
+
+impl<E, DB, P> From<BackfillJob<E, DB, P>> for SingleBlockBackfillJob<E, DB, P> {
+    fn from(value: BackfillJob<E, DB, P>) -> Self {
+        Self {
+            executor: value.executor,
+            provider: value.provider,
+            range: value.range,
+            _db: PhantomData,
+        }
+    }
+}
+
+/// Single block Backfill job started for a specific range.
+///
+/// It implements [`Iterator`] which executes a block each time the
+/// iterator is advanced and yields ([`BlockWithSenders`], [`BlockExecutionOutput`])
+#[derive(Debug)]
+pub struct SingleBlockBackfillJob<E, DB, P> {
+    executor: E,
+    provider: P,
+    range: RangeInclusive<BlockNumber>,
+    _db: PhantomData<DB>,
+}
+
+impl<E, DB, P> Iterator for SingleBlockBackfillJob<E, DB, P>
+where
+    E: BlockExecutorProvider,
+    DB: Database,
+    P: FullProvider<DB>,
```

Comment:
I think this doesn't need fullprovider.

please check the used traits and restrict to those, then we can remove the DB phantom type

Reply:
I would rather do it in a separate PR, the `FullProvider` was in my initial implementation, and it should be removed from both iterators IMO.

Reply:
sg, reasonable